department-of-veterans-affairs/caseflow
# Ki/APPEALS-42518 - All users: Can view correspondence record status

Resolves [APPEALS-42518 - Can view correspondence record status](https://jira.devops.va.gov/browse/APPEALS-42518)

# Description
This updates the value of the "Record status: " to populate either "Pending" or "Completed" depending on whether there is an _active **CorrespondenceRootTask**_.  This also updates some of the controller logic and places "status" on the correspondence model.  

## Acceptance Criteria

## Testing Plan
1. Navigate to Correspondence Cases.
2. Find 2 different correspondence_uuid's from there - 1 still in-progress (pending), and 1 completed.
3. Navigate to the Correspondence Details page for each of the above
    1. i.e. - http://localhost:3000/queue/correspondence/012cd466-b7bd-4f13-b3fd-baa30b8843f8/
5. Validate that the correspondence status matches the status from the details of the correspondence root task itself (pending for all non-complete statuses).  The most surefire way is by using the React tools Components to check the correspondence_tasks belonging to the correspondence you're currently viewing the correspondence details page of:
![Screenshot 2024-07-22 at 2 24 58 PM](https://github.com/user-attachments/assets/8075d4c5-d4ec-488b-b2a5-efbe09510e4b)
![Screenshot 2024-07-22 at 2 26 12 PM](https://github.com/user-attachments/assets/3ed54a6a-1204-4d24-a1aa-4c24f5d8faef)

# Frontend
## User Facing Changes
![Screenshot 2024-07-22 at 2 26 40 PM](https://github.com/user-attachments/assets/75345428-a256-409b-bbba-709b997883cc)
![Screenshot 2024-07-22 at 2 26 30 PM](https://github.com/user-attachments/assets/822c4c56-c033-4a03-9787-1ead8c7fa02f)

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
